sql/src/main/java/io/crate/operation/collect/RowsCollector.java

```diff
@@ -27,48 +27,65 @@
 import io.crate.operation.projectors.RowReceiver;
 
 import javax.annotation.Nullable;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
 
 public class RowsCollector implements CrateCollector {
 
-    private final IterableRowEmitter emitter;
+    private final RowReceiver rowDownstream;
+    private final CompletableFuture<? extends Iterable<?>> future;
+    private final Function<Iterable, Iterable<? extends Row>> transformFunction;
```

Comment:
Could the naming be improved a bit? 
Maybe  `transformFunction -> dataIterableToRows`